std/array.d

```diff
@@ -3852,3 +3852,26 @@ unittest
     assert(appS.data == "hellow");
     assert(appA.data == "hellow");
 }
+
+/++
+Return `.ptr` or null for an array, mitigating the fact we can't use `.ptr` in @safe code.
+
```

Comment:
Maybe add an explanatory note that `a.ptr` could be an invalid but non-null address if `a` was produced by slicing another array `a = b[$ .. $]`, so it's clear to readers that this function is doing something that might prevent memory corruption and isn't just jumping through hoops to appease the compiler.

Reply:
done